File: tasks/task03/Prime.rs
use std::io;

fn is6k1(num: i32) -> bool {
    if num <= 1 {
        return false;
    }
    if num <= 3 {
        return true;
    }
    if num % 2 == 0 || num % 3 == 0 {
        return false;
    }

    let mut i = 5;
    while i * i <= num {
        if num % i == 0 || num % (i + 2) == 0 {
            return false;
        }
        i += 6;
    }
    true
}

fn main() {
    println!("Enter a number: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: i32 = input.trim().parse().expect("Invalid input. Please enter a valid number.");

    println!("Prime numbers up to {} are:", n);
    for i in 2..=n {
        if is6k1(i) {
            println!("{}", i);
        }
    }
}

File: tasks/Task03/Prime.py
def is_6k_1(num):
    if num <= 1:
        return False
    if num <= 3:
        return True
    if num % 2 == 0 or num % 3 == 0:
        return False
    i = 5
    while i * i <= num:
        if num % i == 0 or num % (i + 2) == 0:
            return False
        i += 6
    return True

n = int(input("Enter "))
print("Prime till", n, "are:")
for i in range(2, n + 1):
    if is_6k_1(i):
        print(i)
